src/app/index.ts

```diff
@@ -185,12 +188,15 @@ module.exports = class extends yo {
           name: 'scriptType',
           type: 'list',
           message: 'Choose a script type:',
-          choices: [typescript, javascript],
-          default: typescript,
-          when: !this.options.js && !this.options.ts && !isManifestProject
+          choices: jsonData.getSupportedScripts(projectType),
```

Comment:
Should get the collection once and the reference it in each of the necessary places.